test/Microsoft.AspNet.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -120,7 +120,39 @@ public Task RemoteIPv6Address()
                 client.DefaultRequestHeaders.Connection.Clear();
                 client.DefaultRequestHeaders.Connection.Add("close");
 
-                var response = await client.GetAsync("http://localhost:8791/");
+                var response = await client.GetAsync("http://localhost:8794/");
+                response.EnsureSuccessStatusCode();
+            }
+        }
+
+        [ConditionalFact]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Test hangs after execution on Mono.")]
+        public async Task RequestPathIsNormalized()
+        {
+            var config = new ConfigurationBuilder().AddInMemoryCollection(
+                new Dictionary<string, string> {
+                    { "server.urls", "http://localhost:8795" }
+                }).Build();
+
+            var builder = new WebApplicationBuilder()
+                .UseConfiguration(config)
+                .UseServer("Microsoft.AspNet.Server.Kestrel")
+                .Configure(app =>
+                {
+                    app.Run(async context =>
+                    {
+                        var connection = context.Connection;
+                        Assert.Equal("/\u00C5", context.Request.Path.Value);
+                        await context.Response.WriteAsync("hello, world");
+                    });
+                });
+
+            using (var app = builder.Build())
+            using (var client = new HttpClient())
+            {
+                app.Start();
+
+                var response = await client.GetAsync("http://localhost:8795/A/../%41%CC%8A");
```

Comment:
You can't use HttpClient to test these as it will normalize them before sending them.